### PR TITLE
expose device name and bus info

### PR DESCRIPTION
**Pull request by bazile-clyde**
From the kernel docs [here](https://www.kernel.org/doc/html/v5.5/media/uapi/v4l/vidioc-querycap.html#c.v4l2_capability)
> [card] Name of the device, a NUL-terminated UTF-8 string. For example: “Yoyodyne TV/FM”. One driver may support different brands or models of video hardware. This information is intended for users, for example in a menu of available devices. Since multiple TV cards of the same brand may be installed which are supported by the same driver, **this name should be combined with the character device file name (e. g. /dev/video2) or the bus_info string to avoid ambiguities**.

> [bus_info] Location of the device in the system, a NUL-terminated ASCII string. For example: “PCI:0000:05:06.0”. **This information is intended for users, to distinguish multiple identical devices**. If no such information is available the field must simply count the devices controlled by the driver (“platform:vivi-000”). The bus_info must start with “PCI:” for PCI boards, “PCIe:” for PCI Express boards, “usb-” for USB devices, “I2C:” for i2c devices, “ISA:” for ISA devices, “parport” for parallel port devices and “platform:” for platform devices.

### CODE REVIEW

File: webcam.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func Open(path string) (*Webcam, error) {
 	}
 
 	w := new(Webcam)
-	w.fd = uintptr(fd)
+	w.fd = fd
```
Comment (bazile-clyde) [inline webcam.go:57]: nit: this already done above.

File: webcam.go [hunk 2]
```diff
@@ -86,6 +86,16 @@ func (w *Webcam) GetSupportedFormats() map[PixelFormat]string {
 	return result
 }
 
+// GetName returns the human-readable name of the device
+func (w *Webcam) GetName() (string, error) {
+	return getName(w.fd)
+}
+
+// GetBusInfo returns the location of the device in the system
+func (w *Webcam) GetBusInfo() (string, error) {
```
Comment (edaniels) [inline webcam.go:95]: should this have a better name in this case? Like path or something? the syscall side is fine but this file is an opportunity for cleaner names

File: v4l2.go [hunk 3]
```diff
@@ -311,6 +311,24 @@ func getFrameSize(fd uintptr, index uint32, code uint32) (frameSize FrameSize, e
 	return
 }
 
+func getName(fd uintptr) (string, error) {
+	caps := &v4l2_capability{}
```
Comment (edaniels) [inline v4l2.go:315]:
```suggestion
	var caps v4l2_capability
```

and then do pointer to below

File: v4l2.go [hunk 4]
```diff
@@ -311,6 +311,24 @@ func getFrameSize(fd uintptr, index uint32, code uint32) (frameSize FrameSize, e
...
+	if err := ioctl.Ioctl(fd, VIDIOC_QUERYCAP, uintptr(unsafe.Pointer(caps))); err != nil {
+		return "", err
+	}
+
+	return CToGoString(caps.card[:]), nil
+}
+
+func getBusInfo(fd uintptr) (string, error) {
+	caps := &v4l2_capability{}
```
Comment (edaniels) [inline v4l2.go:324]:
```suggestion
	var caps v4l2_capability
```

**Review by edaniels — CHANGES_REQUESTED** [review]

File: v4l2.go (hunk 3 above)
Comment (bazile-clyde) [reply v4l2.go:315]: I opted to keep things consistent with the other code in this file. I don't think this changes the behavior at all but let me know if that isn't true.

Comment (edaniels) [reply v4l2.go:317]: Let's go with the suggested change since it's better/common style.

**Comment by edaniels** [comment]
LGTM

**Pull request merged**